src/controllers/TalksController.php

```diff
@@ -498,12 +498,89 @@ protected function getTalkDataFromRequest(PDO $db, Request $request, $event_id)
             FILTER_SANITIZE_URL
         );
 
-        $talk['speakers'] = array_map(function ($speaker) {
-            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
-            $speaker = trim($speaker);
-            return $speaker;
-        }, (array) $request->getParameter('speakers'));
+        $talk['speakers'] = array_map(
+            function ($speaker) {
+                $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                $speaker = trim($speaker);
+                return $speaker;
+            },
+            (array) $request->getParameter('speakers')
+        );
+
+        return $talk;
+    }
+
+    public function getSpeakersForTalk(Request $request, PDO $db)
+    {
+        $talk_id = $this->getItemId($request);
+        $talk = $this->getTalkById($db, $request, $talk_id);
+        $collection = new TalkModelCollection([$talk], 1);
+        return $collection->getTalks()[0]->speakers;
+    }
+
+    public function linkUserToTalk(Request $request, PDO $db)
```

Comment:
I did spend quite a bit of time over the naming of this - as per my message in IRC, either party can 'start' the linking process (the speaker can claim or the host can assign) - as such, I do think this 'setSpeakerForTalk' is a better name